app/Components/Game/index.js

```diff
@@ -2,64 +2,71 @@ import React from 'react';
 import ChatApp from '../ChatApp';
 import Profile from '../Profile';
 import Quiz from '../Quiz';
-
+import RoleSelectionMenu from '../RoleSelectionMenu';
 import io from 'socket.io-client';
 let socket = io.connect('');
  
 class Game extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {at_capacity: false, role: null, round_over: false, is_active_game: false};
-    socket.on('connect', () => this._connect());
-    socket.on('init', (username) =>this._initialize(username));
-    socket.on('getroom', () => this._get_room());
-    socket.on('adduser', (user_role) => this._add_user(user_role));
-    socket.on('fullhouse', () => this._full_house());
-    socket.on('isgameID', (flag) => this._is_game_ID(flag));
+    this.state = {at_capacity: false, role: null, round_over: false, is_active_game: false, takenRoles: null};
+    socket.on('isgameID', (flag, takenRoles) => this._is_game_ID(flag, takenRoles));
```

Comment:
In general, when you're adding event handlers like this, do it in `componentDidMount` so that it only runs when React actually mounts the component on the page.  And then in `componentWillUnmount` you'll want to make sure it stops listening to that event too (otherwise the event handler stays bound and keep getting triggered even after the React component is gone).  The React [docs](https://facebook.github.io/react/docs/react-component.html) have more info on suggestions for how to use the lifecycle hooks too if you're curious.